src/core/search/mrmw_mutex_test.cc

```diff
@@ -0,0 +1,180 @@
+// Copyright 2025, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/mrmw_mutex.h"
+
+#include <random>
+#include <thread>
+
+#include "base/gtest.h"
+#include "base/logging.h"
+
+namespace dfly::search {
+
+namespace {
+
+// Helper function to simulate reading operation
+void ReadTask(MRMWMutex* mutex, std::atomic<size_t>& read_count, size_t sleep_time) {
+  read_count.fetch_add(1, std::memory_order_relaxed);
+  MRMWMutexLock lock(mutex, MRMWMutex::LockMode::kReadLock);
```

Comment:
I would really like to know how this works and does not crash. You use fiber synchronization primitives in a non proactor context (nothing is setup in the test). Something seems wrong here 🤔